src/main.rs

```diff
@@ -1,112 +1,176 @@
+    } else {
+        format!("./{}/{}/{}", project_name, sub_dir, name)
+    };
+
+    File::create(&file_dir)?;
+
+    if name == "main.c" {
+        write_main_c(&file_dir);
+    } else if name == "Makefile" {
+        write_makefile(&file_dir);
+    } else if name == "README.md" {
+        write_readme(project_name, &file_dir);
+    } else {
+        write_default_content(&file_dir);
     }
+
+    Ok(())
 }
 
-fn create_project() -> Result<(), Box<dyn std::error::Error>> {
-    let cinit = Cinit::new();
+fn write_main_c(file_dir: &str) {
+    let content =
+        "#include <stdio.h>\n\nint main() {\n    printf(\"Hello, world!\\n\");\n    return 0;\n}\n";
+    let _ = fs::write(file_dir, content);
+    println!("Content written to {}", file_dir);
+}
 
-    cinit.create_sub_directory("src")?;
+fn write_makefile(file_dir: &str) {
+    let content = "CC = gcc\nCFLAGS = -Wall\n\nmain:\n\t$(CC) $(CFLAGS) -o main ./src/main.c\n";
+    let _ = fs::write(file_dir, content);
+    println!("Content written to {}", file_dir);
+}
 
-    if let Some(file_name) = &cinit.0.lib {
-        cinit.create_lib_folder(file_name)?;
-    }
+fn write_readme(project_name: &String, file_dir: &str) {
+    let content = format!(
+            "# {}\n\nA simple C project generated with cinit.\n\n## Structure\n\n- `main.c`: Entry point of the program.\n- `Makefile`: Build system configuration.\n- `README.md`: Project description.\n\n## Build\n\nTo build the project, run:\n\n```bash\nmake\n```\n\n## Run\n\nAfter building, run the program:\n\n```bash\n./main\n```\n",
+            project_name
+        );
+    let _ = fs::write(file_dir, content);
+    println!("README.md created and content written to {}", file_dir);
+}
 
-    cinit.create_file("src", "main.c")?;
-    cinit.create_file("", "README.md")?;
-    cinit.create_file("", "Makefile")?;
+fn write_default_content(file_dir: &str) {
+    let content = "// Default content\n";
+    let _ = fs::write(file_dir, content);
+    println!("Content written to {}", file_dir);
+}
 
+fn create_lib_folder(
+    project_name: &String,
+    file_name: &str,
+) -> Result<(), Box<dyn std::error::Error>> {
+    create_sub_directory(project_name, "lib")?;
+    create_file(project_name, "lib", format!("{file_name}.c").as_str())?;
+    create_file(project_name, "lib", format!("{file_name}.h").as_str())?;
     Ok(())
 }
 
-fn main() {
-    match create_project() {
-        Ok(_) => println!("Enjoy coding :)"),
-        Err(err) => eprintln!("ERROR: {}", err),
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    if let Some(command) = Cinit::new().0.command {
+        match command {
+            Commands::CNew { project_name, lib } => {
+                c_new(&project_name, lib)?;
+            }
+            Commands::BuildC => {
+                build_c()?;
+                //println!("{:?}", std::env::current_dir())
+            }
+            Commands::BulidCRunC => eprintln!("Developing"),
```

Comment:
I think you can just use `todo!()` macro if this feature is not developed yet and the compiler will shut the fuck up